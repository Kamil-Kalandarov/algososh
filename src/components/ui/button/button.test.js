import renderer from "react-test-renderer";
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

describe('test Button component', () => {
    it('render button with text', () => {
        const tree = renderer.create(<Button text={'test text'}/>).toJSON()
        expect(tree).toMatchSnapshot()
    });
    it('render button without text', () => {
        const tree = renderer.create(<Button />).toJSON()
        expect(tree).toMatchSnapshot()
    });
    it('render disabled button', () => {
        const tree = renderer.create(<Button disabled={true}/>).toJSON()
        expect(tree).toMatchSnapshot()
    });
    it('render loading button', () => {
        const tree = renderer.create(<Button isLoader={true}/>)
        expect(tree).toMatchSnapshot()
    });
    const someFunction = () => {
        alert('test text')
    }
    it('button click', () => {
        window.alert = jest.fn()
        const tree = render(<Button onClick={someFunction} text={'test text'}/>)
        const button = screen.getByText('test text')
        fireEvent.click(button)
        expect(window.alert).toHaveBeenCalledWith('test text')
    })
})