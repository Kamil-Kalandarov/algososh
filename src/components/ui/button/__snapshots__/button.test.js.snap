// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component render button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;

exports[`test Button component render button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`test Button component render disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`test Button component render loading button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
